tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -223,11 +229,12 @@ impl<S: Subscriber> Layer<S> for EnvFilter {
 
     fn enabled(&self, metadata: &Metadata<'_>, _: Context<'_, S>) -> bool {
         let level = metadata.level();
+        let target = metadata.target();
         self.scope
             .with(|scope| {
                 for filter in scope.iter() {
-                    if filter >= level {
-                        return true;
+                    if &filter.level >= level {
+                        return filter.target.matches(&target);
```

Comment:
Again, as I mentioned before in https://github.com/tokio-rs/tracing/pull/368#discussion_r331154920, this should only return if the target matches. Otherwise, we need to continue iterating.
```suggestion
                        if filter.target.matches(&target) {
                            return true;
                        ]
```